fluent-react/src/markup.js

```diff
@@ -1,6 +1,7 @@
 /* eslint-env browser */
 
-const TEMPLATE = document.createElement("template");
+const TEMPLATE = typeof(document) !== "undefined" ?
+  document.createElement("template") : null;
 
 export function parseMarkup(str) {
```

Comment:
I'd like to provide a meaningful error message in case we reach the default `parseMarkup` but `document` is `undefined`. In the current PR, if no custom markup parser function is given by the developer, the default `parseMarkup` will throw a `TypeError` trying to assign `str` to `null.innerHTML`.

Rather than check if `TEMPLATE === null` in every call to the default `parseMarkup`, I suggest to go one level up on the ladder of abstraction and define a markup parser _creator_. Here's a default one which is defensive; something like this:

```js
function createParseMarkup() {
  if (typeof(document) === "undefined") {
    // We can't use <template> to sanitize translations.
    throw new Error(
      "Without `document` translations cannot be safely sanitized. " +
      "Pass a custom `function parseMarkup(str: string): string;` prop " +
      "to the <LocalizationProvider>."
    );
  }

  const template = document.createElement("template");

  return function parseMarkup(str) {
    template.innerHTML = str;
    return template.content;
  };
}
```